runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeOfferSlot.java

```diff
@@ -28,38 +28,55 @@
 import java.awt.BorderLayout;
 import java.awt.CardLayout;
 import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.GridLayout;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
 import java.awt.image.BufferedImage;
+import java.io.IOException;
 import javax.annotation.Nullable;
-import javax.swing.BorderFactory;
-import javax.swing.Box;
-import javax.swing.BoxLayout;
+import javax.imageio.ImageIO;
 import javax.swing.ImageIcon;
 import javax.swing.JLabel;
 import javax.swing.JPanel;
-import javax.swing.JProgressBar;
-import javax.swing.border.TitledBorder;
+import javax.swing.border.EmptyBorder;
 import lombok.extern.slf4j.Slf4j;
 import net.runelite.api.GrandExchangeOffer;
 import net.runelite.api.GrandExchangeOfferState;
+import static net.runelite.api.GrandExchangeOfferState.CANCELLED_BUY;
+import static net.runelite.api.GrandExchangeOfferState.CANCELLED_SELL;
 import static net.runelite.api.GrandExchangeOfferState.EMPTY;
 import net.runelite.api.ItemComposition;
+import net.runelite.client.ui.ColorScheme;
+import net.runelite.client.ui.FontManager;
+import net.runelite.client.ui.components.ThinProgressBar;
 import net.runelite.client.util.StackFormatter;
 
 @Slf4j
 public class GrandExchangeOfferSlot extends JPanel
 {
-	private static final Color GE_INPROGRESS_ORANGE = new Color(0xd8, 0x80, 0x20).brighter();
-	private static final Color GE_FINISHED_GREEN = new Color(0, 0x5f, 0);
-	private static final Color GE_CANCELLED_RED = new Color(0x8f, 0, 0);
 
-	private static final String INFO_CARD = "INFO_CARD";
-	private static final String EMPTY_CARD = "EMPTY_CARD";
+	private static final String FACE_CARD = "FACE_CARD";
+	private static final String DETAILS_CARD = "DETAILS_CARD";
 
+	private JPanel container = new JPanel();
 	private final CardLayout cardLayout = new CardLayout();
+
 	private final JLabel itemIcon = new JLabel();
-	private final TitledBorder itemName = BorderFactory.createTitledBorder("Nothing");
-	private final JLabel offerState = new JLabel("Text so the label has height");
-	private final JProgressBar progressBar = new JProgressBar();
+	private final JLabel itemName = new JLabel();
+	private final JLabel offerInfo = new JLabel();
+	private final JLabel switchFaceViewIcon = new JLabel();
+
+	private final JLabel itemPrice = new JLabel();
+	private final JLabel offerSpent = new JLabel();
+	private final JLabel switchDetailsViewIcon = new JLabel();
+
+	private final ThinProgressBar progressBar = new ThinProgressBar();
+
+	private ImageIcon rightArrowIcon;
```

Comment:
Move initialization of both icons to static { } initializer and make them constants